### PR TITLE
voting: Implement missing try-catch in VotingVoteDialog::vote

### LINKED ISSUE #1979: Wallet Crashes to desktop when voting on expired poll.
This was reported on CCT forum by Typv0deister and I tested it and was able to replicate the problem.

Image  of error:  [https://i.gyazo.com/2381dd88a6a07c3c0f4012aa39b61041.png](https://i.gyazo.com/2381dd88a6a07c3c0f4012aa39b61041.png)

Forum post: [https://cryptocurrencytalk.com/topic/169561-gridcoin-client-crash-on-voting/](https://cryptocurrencytalk.com/topic/169561-gridcoin-client-crash-on-voting/)

replicated on Wallet version 5.1.0.0

### PR DESCRIPTION
**Pull request by jamescowens**
The VotingVoteDialog::vote method is missing a try-catch. It is not obvious, but the VoteBuilder::ForPoll constructor/assignment has a throw for VotingError in it. You cannot put a tightly scoped try-catch around the VoteBuilder builder = VoteBuilder::ForPoll(*poll, ref->Txid()); assignment, because the builder is used later in the vote function. So an overall try-catch around the entire thing is necessary to ensure this error is caught here and not allowed to bubble up to the main error handler in bitcoin.cpp (which ends program execution in a "runaway exception").

This closes #1979.

### CODE REVIEW

File: src/qt/votingdialog.cpp [hunk 1]
```diff
@@ -885,53 +885,61 @@ void VotingVoteDialog::resetData(const VotingItem *item)
...
-                builder = builder.AddResponse(row);
+        try {
+            for (int row = 0; row < answerList_->count(); ++row) {
+                if (answerList_->item(row)->checkState() == Qt::Checked) {
+                    builder = builder.AddResponse(row);
+                }
             }
+        } catch (const VotingError& e) {
+            voteNote_->setText(e.what());
+            return;
```
Comment (cyrossignol) [inline src/qt/votingdialog.cpp]: Want to drop this `try`...`catch`? It looks redundant now.

File: src/qt/votingdialog.cpp [hunk 2]
```diff
@@ -885,53 +885,61 @@ void VotingVoteDialog::resetData(const VotingItem *item)
...
 
-    try {
-        SendVoteContract(std::move(builder));
-    } catch (const VotingError& e) {
+        try {
+            SendVoteContract(std::move(builder));
+        } catch (const VotingError& e) {
+            voteNote_->setText(e.what());
+            return;
+        }
```
Comment (cyrossignol) [inline src/qt/votingdialog.cpp]: Same here.

File: src/qt/votingdialog.cpp (hunk 1 above)
Comment (jamescowens) [reply src/qt/votingdialog.cpp]: Yep.

File: src/qt/votingdialog.cpp (hunk 2 above)
Comment (jamescowens) [reply src/qt/votingdialog.cpp]: Yep.

**Review by cyrossignol — APPROVED** [review]

**Pull request merged**